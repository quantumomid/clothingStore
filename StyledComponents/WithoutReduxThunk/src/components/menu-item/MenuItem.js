import { withRouter } from 'react-router-dom'
import { BackgroundImage, ContentContainer, ContentSubtitle, ContentTitle, MenuItemContainer } from './menuItemStyles'
// import './MenuItem.scss'

function MenuItem({ title, imageUrl, size, history, linkUrl, match }){
    return(
        <MenuItemContainer size={size} onClick={() => history.push(`${match.url}${linkUrl}`)}>   
            <BackgroundImage imageUrl={imageUrl} className="background-image"/>
            <ContentContainer>
                <ContentTitle>{title.toUpperCase()}</ContentTitle>
                <ContentSubtitle>SHOP NOW</ContentSubtitle>
            </ ContentContainer>
        </MenuItemContainer>
    )
}

export default withRouter(MenuItem)