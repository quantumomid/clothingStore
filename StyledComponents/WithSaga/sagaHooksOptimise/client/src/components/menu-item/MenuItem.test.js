import React from "react";
import { shallow } from "enzyme";
import { MenuItem } from "./MenuItem";

describe("Menu Item Component", () => {
    let wrapper;
    let mockMatch;
    let mockHistory;

    beforeEach(() => {

        mockMatch = {
            url: "/shop"
        }

        mockHistory = {
            push: jest.fn()
        }

        const mockProps = {
            title: "hats", 
            imageUrl: "testimage", 
            size: "large", 
            history: mockHistory, 
            linkUrl: "/hats", 
            match: mockMatch
        }

        wrapper = shallow(<MenuItem { ...mockProps } />);
    });

    it("Menu Item Component should render", () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('Should pass size to MenuItemContainer as the prop size', () => {
        expect(wrapper.find('MenuItemContainer').prop('size')).toBe("large");
    });

    it('should pass imageUrl to BackgroundImageContainer as the prop imageUrl', () => {
        expect(wrapper.find('BackgroundImageContainer').prop('imageUrl')).toBe("testimage");
    });

    it('should call history.push with the right string when MenuItemContainer clicked', () => {
        wrapper.find('MenuItemContainer').simulate('click');
        expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}/hats`);
    });
})