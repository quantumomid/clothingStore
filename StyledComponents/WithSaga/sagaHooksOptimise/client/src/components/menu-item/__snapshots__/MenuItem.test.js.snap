// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Item Component Menu Item Component should render 1`] = `ShallowWrapper {}`;
